Stock Entry | Date overlapping issues

### Information about bug

Issue : frappe.exceptions.NameError: Year start date or end date is overlapping with _Test Fiscal Year 2024. To avoid please set company

Failed Testcases :- 
test_stock_manufacture_with_batch_serial_TC_SCK_141
test_stock_manufacture_with_batch_serial_TC_SCK_142
test_stock_manufacture_with_batch_serieal_TC_SCK_140 
test_stock_reco_TC_SCK_127
test_repack_with_additional_costs

### Module

stock

### Version

Frappe Framework: v15.51.2
ERPNext: v15.50.

### Installation method

manual install

### Relevant log output / Stack trace / Full Error Message.

```shell
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/stock/doctype/stock_entry/test_stock_entry.py", line 3797, in test_stock_manufacture_with_batch_serial_TC_SCK_141
    get_fiscal_year(company)
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/stock/doctype/stock_entry/test_stock_entry.py", line 4304, in get_fiscal_year
    create_fiscal_with_company(company)
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/stock/doctype/stock_entry/test_stock_entry.py", line 4295, in create_fiscal_with_company
    fy_doc.submit()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 1060, in submit
    return self._submit()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 1043, in _submit
    return self.save()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 342, in save
    return self._save(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 364, in _save
    return self.insert()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 295, in insert
    self.run_before_save_methods()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 1106, in run_before_save_methods
    self.run_method("validate")
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 974, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 1334, in composer
    return composed(self, method, *args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 1316, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 971, in fn
    return method_object(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/accounts/doctype/fiscal_year/fiscal_year.py", line 34, in validate
    self.validate_overlap()
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/accounts/doctype/fiscal_year/fiscal_year.py", line 110, in validate_overlap
    frappe.throw(
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 603, in throw
    msgprint(
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 568, in msgprint
    _raise_exception()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 519, in _raise_exception
    raise exc
frappe.exceptions.NameError: Year start date or end date is overlapping with _Test Fiscal Year 2024. To avoid please set company
```

@Abhishek8848 

all passed except one :

``` 
FAIL: test_repack_with_additional_costs (erpnext.stock.doctype.stock_entry.test_stock_entry.TestStockEntry)

Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/stock/doctype/stock_entry/test_stock_entry.py", line 516, in test_repack_with_additional_costs
    self.check_gl_entries(
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/stock/doctype/stock_entry/test_stock_entry.py", line 557, in check_gl_entries
    self.assertEqual(expected_gl_entries[i][0], gle[0])
AssertionError: 'Stock In Hand - TCP1' != 'Stock Adjustment - TCP1'
- Stock In Hand - TCP1
+ Stock Adjustment - TCP1
 ```

--- a/erpnext/stock/doctype/material_request/test_material_request.py
+++ b/erpnext/stock/doctype/material_request/test_material_request.py
@@ -5500,7 +5500,7 @@ class TestMaterialRequest(FrappeTestCase):
 	def test_mr_to_2po_to_2pi_sr_partail_return_TC_SCK_114(self):
 		create_company()
 		create_fiscal_year()
-		supplier = create_supplier(supplier_name="_Test Supplier MR")
+		supplier = create_supplier(supplier_name="_Test Supplier MR",default_currency="INR")
 		warehouse = create_warehouse("_Test warehouse PO", company="_Test Company MR")
 		item = item_create("_Test MR")
 		cost_center = frappe.db.get_value("Company","_Test Company MR","cost_center")

--- a/erpnext/stock/doctype/stock_entry/test_stock_entry.py
+++ b/erpnext/stock/doctype/stock_entry/test_stock_entry.py
@@ -444,6 +444,15 @@ class TestStockEntry(FrappeTestCase):
 		self.assertFalse(gl_entries)
 
 	def test_repack_with_additional_costs(self):
+		from erpnext.stock.doctype.item.test_item import create_item
+		create_item("_Test Item")
+		create_item("_Test Item Home Desktop 100")
+		test_records = frappe.get_test_records("Company")
+		test_records = test_records[2:]
+		for rec in test_records:
+			if not frappe.db.exists("Company", rec.get("company_name")):
+				rec["doctype"] = "Company"
+				frappe.get_doc(rec).insert()
 		company = frappe.db.get_value("Warehouse", "Stores - TCP1", "company")
 		create_fiscal_with_company(company)
 		make_stock_entry(

